### PR TITLE
Update and Delete Form API endpoints

### PR DESCRIPTION
**Pull request by jaipise**
## Update and Delete Form API endpoints
Issue Number(s): #57 

What does this PR change and why?
### Checklist
- [x] Requirements have been implemented
- [x] Acceptance criteria is met
- [x] Database schema [docs](https://www.notion.so/gtbitsofgood/Database-Schema-Docs-a841a38d02db4428a7cdc85dcbb8a35c?pvs=4) have been updated or are not necessary
- [x] Code follows design and style guidelines
- [x] Commits follow guidelines (concise, squashed, etc)
- [x] Relevant reviewers (EM) have been assigned to this PR

### How To Test
The steps to test the functionality of your ticket. Include edge cases and testing of error handling, if applicable.

### CODE REVIEW

File: server/db/actions/Form.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    throw new Error("Unable to update form");
+  }
+}
+
+export async function deleteForm(formId) {
+  try {
+    await dbConnect();
+
+    const deletedForm = await Form.findByIdAndDelete(formId);
+    if (!deletedForm) {
```
Comment (afazio1) [inline server/db/actions/Form.js:25]: this is caught in the `catch` and the message gets overridden

**Review by afazio1 — CHANGES_REQUESTED** [review]
great work! Just one minor error handling thing

File: server/db/actions/Form.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  } catch (e) {
+    throw new Error("Unable to update form at this time, please try again");
+  }
+
+  try {
+    return Form.findByIdAndUpdate(formId, formData, {
+      returnDocument: "after",
+    });
+  } catch (e) {
+    throw new Error("Unable to update form");
```
Comment (afazio1) [inline server/db/actions/Form.js:18]: this is caught in the `catch` and the message gets overridden

**Review by afazio1 — CHANGES_REQUESTED** [review]

File: server/db/actions/Form.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  }
+}
+
+export async function deleteForm(formId) {
+  try {
+    await dbConnect();
+
+    const deletedForm = await Form.findByIdAndDelete(formId);
+    if (!deletedForm) {
+      throw new Error("Invalid form ID");
```
Comment (afazio1) [inline server/db/actions/Form.js:28]: this error is caught in the `catch` and the message is overridden

**Review by afazio1 — CHANGES_REQUESTED** [review]

**Review by afazio1 — APPROVED** [review]

File: src/pages/api/forms/[id].js [hunk 4]
```diff
@@ -1,7 +1,117 @@
...
+          return res.status(500).send({
+            success: false,
+            message: "Unable to delete form because form ID does not exist",
+          });
+        }
+
+        return res.status(500).send({
+          success: false,
+          message: "Unable to delete form, please try again",
+          error: error.message,
```
Comment (afazio1) [inline src/pages/api/forms/[id].js]: there shouldn't be `error` key here

**Review by afazio1 — CHANGES_REQUESTED** [review]
found a small thing with response shape

**Comment by jaipise** [comment]
> found a small thing with response shape

Yep sorry, not sure why I added that

**Review by afazio1 — APPROVED** [review]
